rct229/rules/section5/section5rule45.py

```diff
@@ -16,57 +17,81 @@ class Section5Rule45(RuleDefinitionListIndexedBase):
+                    required_fields={"infiltration": ["multiplier_schedule"]},
+                )
 
-            return compare_schedules(
-                infiltration_b["multiplier_schedule"],
-                infiltration_p["multiplier_schedule"],
-                mask_schedule,
-                1.0,
-                data["is_leap_year"],
-            )
+            def get_calc_vals(self, context, data=None):
+                zone_b = context.baseline
+                zone_p = context.proposed
 
-        def rule_check(self, context, calc_vals=None, data=None):
-            total_hours_compared = calc_vals["total_hours_compared"]
-            total_hours_match = calc_vals["total_hours_matched"]
-            # total_hours_compared needs to be greater than 0.0
-            return (
-                total_hours_compared > 0.0 and total_hours_compared == total_hours_match
-            )
+                schedules_b = data["schedules_b"]
+                schedules_p = data["schedules_p"]
+
+                mask_schedule = [1.0] * (
+                    LEAP_YEAR_HOURS if data["is_leap_year"] else REGULAR_YEAR_HOURS
+                )
+
+                infiltration_b = zone_b["infiltration"]
+                infiltration_p = zone_p["infiltration"]
+
+                return compare_schedules(
+                    # search for the schedule used in infiltration data group.
+                    # Set to empty list if non-matched,
+                    # raise exception if not hourly schedule or the hourly_values key is missing
+                    next((getattr_(sched, "schedule", "hourly_values") for sched in schedules_b if sched["id"] == infiltration_b["multiplier_schedule"]), []),
+                    next((getattr_(sched, "schedule", "hourly_values") for sched in schedules_p if sched["id"] == infiltration_p["multiplier_schedule"]), []),
+                    mask_schedule,
+                    1.0,
+                    data["is_leap_year"],
+                )
+
+            def rule_check(self, context, calc_vals=None, data=None):
+                total_hours_compared = calc_vals["total_hours_compared"]
+                total_hours_match = calc_vals["total_hours_matched"]
+                # total_hours_compared needs to be greater than 0.0
+                return (
+                    total_hours_compared > 0.0 and total_hours_compared == total_hours_match
+                )
 
-        def get_fail_msg(self, context, calc_vals=None, data=None):
-            elfh_difference = calc_vals["elfh_difference"]
-            return f"Baseline and proposed infiltration schedules are not the same. The difference of baseline EFLH to proposed EFLH is equal to {elfh_difference}."
+            def get_fail_msg(self, context, calc_vals=None, data=None):
+                elfh_difference = calc_vals["eflh_difference"]
+                return f"Baseline and proposed infiltration schedules are not the same. " \
```

Comment:
Please update the RDS to reflect this message wording change.
BTW: I think it should be "... The difference between baseline EFLH and proposed EFLH is ..."

Reply:
You are correct. Updated